src/NLog/Targets/FileTarget.cs

```diff
@@ -615,7 +586,7 @@ protected override void CloseTarget()
         {
             base.CloseTarget();
 
-            foreach (string fileName in new List<string>(this.initializedFiles.Keys))
+            foreach (string fileName in initializedFiles.GetItems())
```

Comment:
I simplified the implementation of initializedFiles.GetItems() as it was unnecessarily complex but it might still cause the same issue. Unfortunately, I need the advice of a more experienced developer.